cmd/grumble/server.go

```diff
@@ -256,27 +263,28 @@ func (server *Server) handleIncomingClient(conn net.Conn) (err error) {
 	client.user = nil
 
 	// Extract user's cert hash
-	tlsconn := client.conn.(*tls.Conn)
-	err = tlsconn.Handshake()
-	if err != nil {
-		client.Printf("TLS handshake failed: %v", err)
-		client.Disconnect()
-		return
-	}
+	if tlsconn, ok := client.conn.(*tls.Conn); ok {
```

Comment:
Please add a comment describing why. Maybe "Only consider client certificates for direct connections, not WebSocket connections. We do not support TLS-level client certificates for WebSocket client."